libs/ledger/include/ledger/upow/synergetic_execution_manager_interface.hpp

```diff
@@ -46,36 +48,7 @@ class SynergeticExecutionManagerInterface
   /// @}
 };
 
-inline char const *ToString(SynergeticExecutionManagerInterface::ExecStatus status)
-{
-  using ExecStatus = SynergeticExecutionManagerInterface::ExecStatus;
-
-  char const *text = "Unknown";
-
-  switch (status)
-  {
-  case ExecStatus::SUCCESS:
-    text = "Success";
-    break;
-  case ExecStatus::CONTRACT_NAME_PARSE_FAILURE:
-    text = "Contract name parse failure";
-    break;
-  case ExecStatus::INVALID_CONTRACT_ADDRESS:
-    text = "Invalid contract address";
-    break;
-  case ExecStatus::INVALID_NODE:
-    text = "Invalid node";
-    break;
-  case ExecStatus::INVALID_BLOCK:
-    text = "Invalid block";
-    break;
-  case ExecStatus::CONTRACT_REGISTRATION_FAILED:
-    text = "Contract registration failed";
-    break;
-  }
-
-  return text;
-}
+char const *ToString(SynergeticExecutionManagerInterface::ExecStatus status);
```

Comment:
Could be made `constexpr` if left here.

Reply:
The performance benefit is likely to be negligible. Let's keep executable code out of headers if we can.